src/persist/src/indexed/cache.rs

```diff
@@ -93,6 +94,7 @@ impl<L: Blob> BlobCache<L> {
         if key == Self::META_KEY {
             return Err(format!("cannot write trace batch to meta key: {}", Self::META_KEY).into());
         }
+        debug_assert_eq!(batch.validate(), Ok(()), "{:?}", &batch);
```

Comment:
do you think its maybe simpler to do

```
debug_assert!(batch.validate().is_ok(), "{:?}", &batch);
```

(this is totally a style nit so feel free to tioli)